src/FSharp.Core/prim-types.fs

```diff
@@ -3912,6 +3912,16 @@ namespace Microsoft.FSharp.Collections
     type List<'T> = 
        | ([])  :                  'T list
        | ( :: )  : Head: 'T * Tail: 'T list -> 'T list
+       member private this.CustomHashCode(c:IEqualityComparer) =
```

Comment:
Also a (rather small  potential) issue is that if someone has an extension member for the list which the same name, their code will fail to compile. 
Solution would be to have a custom name for it.

Reply:
Are you sure?
I just tested a clash between a private member and:
- An F# extension
- A C#-style extension method
And neither of them fails to compile, even if the extension uses the same name as the private member.

Reply:
Yes, disregard, I thought both Roslyn and us will not check the access modifier, and only name and types